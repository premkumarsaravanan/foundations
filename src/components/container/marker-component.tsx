import * as React from 'react'
import { Dispatch } from 'redux'
import { CoordinateProps } from '@reapit/elements'
import { appointmentDetailRequestData } from '@/actions/appointment-detail'
import { connect } from 'react-redux'

export type MarkerComponentProps = {
  handleOnClick: () => void
  coordinate: CoordinateProps<any>
}

export const MarkerComponent = ({ coordinate, handleOnClick }: MarkerComponentProps) => {
  const { address1, address2 } = coordinate
  return (
    <div onClick={handleOnClick}>
      <p>{address1}</p>
      <p>{address2}</p>
    </div>
  )
}

export type ownProps = {
  id: string
}

export const mapDispatchToProps = (dispatch: Dispatch, ownProps: ownProps) => ({
  handleOnClick: () => dispatch(appointmentDetailRequestData({ id: ownProps.id }))
})

export const MarkerComponentWithConnect = connect(
  null,
  mapDispatchToProps
)(MarkerComponent)
