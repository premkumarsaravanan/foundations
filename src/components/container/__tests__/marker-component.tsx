import * as React from 'react'
import { shallow } from 'enzyme'
import { MarkerComponent, MarkerComponentProps, mapDispatchToProps } from '../marker-component'

describe('Marker Component', () => {
  it('Should match snapshot', () => {
    const props: MarkerComponentProps = {
      handleOnClick: jest.fn(),
      coordinate: {
        lng: 1,
        lat: 1,
        id: '1',
        address1: '1',
        address2: '1'
      }
    }
    const wrapper = shallow(<MarkerComponent {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('mapDispatchToProps', () => {
    const mockDispatch = jest.fn()
    const mockOwnProps = {
      coordinate: {
        id: '1'
      }
    }
    const { handleOnClick } = mapDispatchToProps(mockDispatch, mockOwnProps)
    handleOnClick()
    expect(mockDispatch).toBeCalledWith({
      data: {
        id: '1'
      },
      type: 'APPOINTMENT_DETAIL_REQUEST_DATA'
    })
  })
})
