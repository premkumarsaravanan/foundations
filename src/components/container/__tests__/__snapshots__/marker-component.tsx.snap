// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker Component Should match snapshot 1`] = `
<div
  onClick={[MockFunction]}
>
  <p>
    1
  </p>
  <p>
    1
  </p>
</div>
`;
